assets/js/data/utils/process-error-response.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import {
+	createNotice,
+	createNoticeIfVisible,
+	DEFAULT_ERROR_MESSAGE,
+} from '@woocommerce/base-utils';
+import { decodeEntities } from '@wordpress/html-entities';
+import { isObject, objectHasProp, ApiErrorResponse } from '@woocommerce/types';
+import { noticeContexts } from '@woocommerce/base-context/event-emit/utils';
+
+type ApiParamError = {
+	param: string;
+	id: string;
+	code: string;
+	message: string;
+};
+
+const isApiResponse = ( response: unknown ): response is ApiErrorResponse => {
+	return (
+		isObject( response ) &&
+		objectHasProp( response, 'code' ) &&
+		objectHasProp( response, 'message' ) &&
+		objectHasProp( response, 'data' )
+	);
+};
+
+/**
+ * Flattens error details which are returned from the API when multiple params are not valid.
+ *
+ * - Codes will be prefixed with the param. For example, `invalid_email` becomes `billing_address_invalid_email`.
+ * - Additional error messages will be flattened alongside the main error message.
+ * - Supports 1 level of nesting.
+ * - Decodes HTML entities in error messages.
+ */
+const getErrorDetails = ( response: ApiErrorResponse ): ApiParamError[] => {
+	const errorDetails = objectHasProp( response.data, 'details' )
+		? Object.entries( response.data.details )
+		: null;
+
+	if ( ! errorDetails ) {
+		return [];
+	}
+
+	return errorDetails.reduce(
+		(
+			acc,
+			[
+				param,
+				{ code, message, additional_errors: additionalErrors = [] },
+			]
+		) => {
+			const details = [
+				...acc,
+				{
+					param,
+					id: `${ param }_${ code }`,
+					code,
+					message: decodeEntities( message ),
+				},
+				...( Array.isArray( additionalErrors )
+					? additionalErrors.flatMap( ( additionalError ) => {
+							if (
+								! objectHasProp( additionalError, 'code' ) ||
+								! objectHasProp( additionalError, 'message' )
+							) {
+								return [];
+							}
+							return [
+								{
+									param,
+									id: `${ param }_${ additionalError.code }`,
+									code: additionalError.code,
+									message: decodeEntities(
+										additionalError.message
+									),
+								},
+							];
+					  } )
+					: [] ),
+			];
+			return details;
+		},
+		[] as ApiParamError[]
+	);
+};
+
+/**
+ * Processes the response for an invalid param error, with response code rest_invalid_param.
+ */
+const processInvalidParamResponse = ( response: ApiErrorResponse ) => {
+	const errorDetails = getErrorDetails( response );
+
+	errorDetails.forEach( ( { code, message, id, param } ) => {
+		switch ( code ) {
+			case 'invalid_phone':
```

Comment:
Future parameters we want to handle should be added here.

Reply:
Could we eventually allow extensions to add their own handlers here?

Reply:
Yes, I think so. We'll work this out in a follow up.